src/rvoice/fluid_rev.c

```diff
@@ -1,517 +1,1313 @@
+static void update_stereo_coefficient(fluid_late * late, fluid_real_t wet1)
 {
-  int i;
-  fluid_real_t* buf = comb->buffer;
-  int len = comb->bufsize;
-  for (i = 0; i < len; i++) {
-    buf[i] = DC_OFFSET; /* This is not 100 % correct. */
-  }
+	int i;
+	for(i = 0; i < NBR_DELAYS; i++)
+	{
+		/*  delay lines output gains vectors Left and Right
+
+                           L    R
+                       0 | 1    1| 
+                       1 |-1    1|
+                       2 | 1   -1| 
+                       3 |-1   -1|
+
+                       4 | 1    1| 
+                       5 |-1    1| 
+         stereo gain = 6 | 1   -1|
+                       7 |-1   -1|
+ 
+                       8 | 1    1| 
+                       9 |-1    1| 
+                       10| 1   -1|
+                       11|-1   -1|
+		*/ 
+      
+		late->out_left_gain[i] = wet1;
+		/*  Sets Left coefficients first */
+		if(i%2) /* Left is 1,-1, 1,-1, 1,-1,.... */
+		{
+			late->out_left_gain[i] *= -1 ;
+		}
+		/* Now sets right gain as function of Left */
+		/* for right line 1,2,5,6,9,10,13,14, right = - left */
+		if((i==1)||(i==2)||(i==5)||(i==6)||(i==9)||(i==10)||(i==13)||(i==14))
+		{
+			/* Right is reverse of Left */
+			late->out_right_gain[i] = -1 * late->out_left_gain[i]; 
+		}
+		else /* for Right : line 0,3,4,7,8,11,12,15 */
+		{   /* Right is same as Left */
+			late->out_right_gain[i] = late->out_left_gain[i];  
+		}
+	}
 }
 
-void
-fluid_comb_setdamp(fluid_comb* comb, fluid_real_t val)
+/*-----------------------------------------------------------------------------
+ fluid_late destructor
+ @param late pointer on late structure.
+-----------------------------------------------------------------------------*/
+static void delete_fluid_rev_late(fluid_late * late)
 {
-  comb->damp1 = val;
-  comb->damp2 = 1 - val;
+	long i;
+	for(i =0; i < NBR_DELAYS; i++)
+	{
+		FLUID_FREE(late->mod_delay_lines[i].dl.line);
+	}
 }
 
-fluid_real_t
-fluid_comb_getdamp(fluid_comb* comb)
+/*-----------------------------------------------------------------------------
+ Creates the fdn reverb
+ @param late, pointer on the fnd late reverb to initialize.
+ @param sample_rate the sample rate.
+ @return FLUID_OK if success, FLUID_FAILED otherwise.
+-----------------------------------------------------------------------------*/
+static int create_fluid_rev_late(fluid_late * late, fluid_real_t sample_rate)
 {
-  return comb->damp1;
+	int result ; /* return value */
+	long i ;
+
+	FLUID_MEMSET(late, 0,  sizeof(fluid_late));
+
+	late->samplerate = sample_rate;
+
+	/*--------------------------------------------------------------------------
+	  First initialize the modulated delay lines 
+	*/
+
+	for(i = 0; i < NBR_DELAYS; i++)
+	{
+		/* sets local delay lines's parameters */
+		result = set_mod_delay_line(&late->mod_delay_lines[i], 
+		                             delay_length[i], MOD_DEPTH, MOD_RATE );
+		if (result == FLUID_FAILED)
+		{
+			delete_fluid_rev_late(late);
```

Comment:
Remove this delete_* call pls. Cleanup should be done by parent constructor function `new_fluid_revmodel()`.

Reply:
All cleanup done.

>Please add NULL-guards to destructor functions: fluid_return_if_fail(rev != NULL);

Testing rev to  not NULL but not using fluid_return_if_fail(), as this is an internal  reverb interface API , not a public API. Hope it is correct ?